consensus/leader_election/module.go

```diff
@@ -55,10 +55,23 @@ func (m *leaderElectionModule) GetBus() modules.Bus {
 }
 
 func (m *leaderElectionModule) ElectNextLeader(message *typesCons.HotstuffMessage) (typesCons.NodeId, error) {
-	return m.electNextLeaderDeterministicRoundRobin(message), nil
+	nodeId, err := m.electNextLeaderDeterministicRoundRobin(message)
+	if err != nil {
+		return typesCons.NodeId(0), err
+	}
+	return nodeId, nil
 }
 
-func (m *leaderElectionModule) electNextLeaderDeterministicRoundRobin(message *typesCons.HotstuffMessage) typesCons.NodeId {
+func (m *leaderElectionModule) electNextLeaderDeterministicRoundRobin(message *typesCons.HotstuffMessage) (typesCons.NodeId, error) {
 	value := int64(message.Height) + int64(message.Round) + int64(message.Step) - 1
-	return typesCons.NodeId(value%int64(len(m.GetBus().GetConsensusModule().ValidatorMap())) + 1)
+
+	uCtx, err := m.GetBus().GetUtilityModule().NewContext(int64(message.Height))
+	if err != nil {
+		return typesCons.NodeId(0), err
+	}
+	vals, err := uCtx.GetPersistenceContext().GetAllValidators(int64(message.Height))
```

Comment:
Let's create a read context `m.GetBus().GetPersistenceModule().NewReadContext(int64(height))` like you did in #374. Keeps it simpler.

Reply:
We are doing it this way because of this
![image](https://user-images.githubusercontent.com/29378614/207464209-e13cf635-99ca-4170-99dd-1ecdc830ddb7.png)

remember we discussed it and you asked me to get the context from Utility?